### PR TITLE
Heidi-Yoon NDVI Code Review 2022

### PR DESCRIPTION
**Pull request by AreteY**
# Pull Request Template

## CI Checks
- [ ] The notebook runs from start to finish on all operating systems:
   - [ ] Mac
   - [ ] Windows
   - [ ] Linux 
### Reproducibility
- [ ] Are the data downloaded in the code
- [ ] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [ ] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [ ] Does the notebook run from start to finish?

### Functions
- [ ] Do functions follow PEP 8 format conventions?
- [ ] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [ ] Are function names expressive (the name describes what the function does)?
- [ ] Are functions easy to understand and read?
- [ ] How many tasks does each function do? (ideally a function does one thing well).
### Package imports
- [ ] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [ ] Are variable names throughout the code, expressive?
- [ ] Suggest changes if not, highlight what was done well
- [ ] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [ ] Are segments of code repeated in the file or is the code DRY?
- [ ] Are loops used to optimize DRY code?
- [ ] Are functions used to optimize DRY code?
- [ ] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [ ] Highlight any novel approaches to completing the assignment.

### CODE REVIEW

**Comment by rami8797** [comment]
Hi @AreteY! Here are some comments on your code. I'll submit an issue as well.

I think that a comment may have been helpful in line 148 to describe what is being done here.
https://github.com/earthlab-education/ea-2021-ndvi-automation-review/blob/9b7ae60a568be82901fbf0e70ebb4dfc38a71867/yoon_heidi_ndvi.py#L148

This is nitpicky, but just to keep things consistent, it might make sense to start at the end of the path or the beginning of the path to access both the date and time information.

https://github.com/earthlab-education/ea-2021-ndvi-automation-review/blob/9b7ae60a568be82901fbf0e70ebb4dfc38a71867/yoon_heidi_ndvi.py#L333-L334

**Comment by AreteY** [comment]
Thanks @rami8797 ! 
-Ah yes, thanks for mentioning the matplotlib converters. Adding a comment would make it better.
-Thanks! I can change the date indices so that I count the same way.

File: yoon_heidi_ndvi.py [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import xarray as xr
+import rioxarray as rxr
+import earthpy as et
+import earthpy.mask as em
+import pyproj
+from matplotlib.dates import DateFormatter
```
Comment (AreteY) [inline yoon_heidi_ndvi.py:146]:
```suggestion
from matplotlib.dates import DateFormatter

# Help with date time conversion between pandas and matplotlib
```

File: yoon_heidi_ndvi.py [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+    site = []
+    column_names = ["mean_ndvi", "site", "date"]
+
+    # For all the dates, list all of the band paths and qa path
+    for adate in all_dates:
+        band_paths = sorted(
+            glob(os.path.join(adate, "*band*[4-5].tif")))
+        landsat_qa_path = glob(os.path.join(adate, "*qa*"))
+        # Store the date and file name for each date.
+        dates.append(adate[-29:-21])
```
Comment (AreteY) [inline yoon_heidi_ndvi.py:333]:
```suggestion
        dates.append(adate[50:58])
```

**Pull request closed without merging**